[IV-21-22] Objetivo 2

morevi opened the pull request:
En este PR trataré de hacer avanzar [M1](https://github.com/francisco3207/IVProyecto/milestone/2).
- Continúa #7
- Closes #9, relacionado con la HU anterior.
- Closes #10 
- Closes #12 

## Lista de comprobación
Sobre la estructura del repositorio

- [x] ¿He seguido las mejores prácticas en nombre de las clases y ficheros y disposición de los mismos?

Sobre el análisis del problema

- [x] ¿Se ha documentado qué análisis se ha hecho sobre el dominio para decir lo que se ha creado?
- [x] ¿Se ha documentado por qué se ha elegido que lo creado sea un objeto valor, una entidad o un agregado?
- [x] ¿He propuesto un producto mínimamente viable, que en general será un solo objeto valor que no dependa de ningún otro (y que sea la base de muchos otros)?.

Sobre la planificación y la programación
- [x] ¿Los issues representan un problema, y no una tarea?
- [x] ¿Todos los mensajes de commit explican el cambio, y no se limitan a repetir el nombre del fichero que se ha cambiado?
- [x] ¿Los mensajes de commit siguen el el formato estándar y las buenas prácticas.
- [x] ¿Se ha hecho una revisión real del código para comprobar que todos los atributos y funciones creadas están respaldadas por una HU?
- [x] ¿Todos los issues creados están asignados a una HU?
- [x] ¿Todos los cambios en el código están asignados a un issue?
- [ ] Los issues sobre los que estoy trabajando, ¿han sido asignados al primer milestone?
- [ ] ¿Se ha asignado al mismo milestone el PR que se ha hecho?
- [x] ¿Son los milestones sobre los que estoy trabajando productos mínimamente viables? ¿O tengo que solicitar al product manager que cree milestones adicionales o precise de qué producto se trata?

morevi committed:
Define the Producto module

This class will contain the information needed to describe the
product that the client will need to sell.

It was added inside a python module just in case we need to add
auxiliar or related code outside of Producto class.

See #9

morevi committed:
Add basic justification

Closes #10

morevi committed:
Add this file to be able to pass the tests

See #9

morevi commented:
Esperando review @francisco3207

morevi committed:
Enforce PEP8

See #9

morevi committed:
Add explanation for class Producto

See #9

morevi commented:
@JJ Listo para revisión!

francisco3207 reviewed (commented):
En calcular_beneficio podrias añadir el precio total segun los meses del año con respecto a otros años.

JJ commented on docs/clase_producto.md:
```diff
@@ -0,0 +1,20 @@
+# Clase producto
+
+Para representar los cultivos, se ha creado una clase _producto_.
```
Esto debe estar en un issue, no aquí.

JJ commented on producto/producto.py:
```diff
@@ -0,0 +1,52 @@
+    tipo, y la fecha en la que se realizo el plantado.
+    '''
+
+    def __init__(self, tipo: str, rendimiento_base: float, fecha_plantado):
```
Estás mezclando la producción, que supongo que sería anual, con la producción anual. O algo así, no lo tengo muy claro. Deberías crear un issue para ver realmente qué es lo que necesita, y qué es lo que va a usarse en la lógica de negocio.

JJ reviewed (changes requested):
El issue #9 no plantea ningún problema de forma realista, y la resolución en una sola clase no sigue el principio de "separation of concerns", una clase debe hacer una sola cosa.

morevi committed:
Remove predecir_rendimiento

Rendimiento is given by the cooperative, it is NOT a calculated
value.

See #9

morevi committed:
Remove fecha_plantado from class constructor

An olive can have 50 years of useful life, there is no reason to
include this information

See #9

morevi committed:
Remove everything that is not in Producto scope

Producto should hold only information about one harvest.
The predictions are not related to any Producto, so it should be
on a separate class

See #9

morevi committed:
Add interface class for Historico

This will act as a log for prices for a specific type of olive,
allowing predictions to be made.

See #12

morevi committed:
Remove extra documentation

This was explained on #9 and it should not be in form of
documentation.

morevi commented:
@JJ Listo para revisión.
Se ha:
- modificado el contenido de #9 para aclarar que se necesita, y creado #12. 
- reformulado el issue como un problema.
- separado en 2 clases con funcionalidad separada.
- eliminado información que creía de utilidad pero ha acabado siendo inutil o un dato de entrada (el rendimiento no lo calculamos nosotros, asi que desaparecen varios atributos y métodos)

JJ reviewed (approved):
El código que no falla es el que no se escribe, así que el enfoque minimalista puede servir, sobre todo si al jefe le parece bien. Va a ser más trabajo para @francisco3207 , sobre todo en el objetivo 4...

JJ commented:
A ver, @morevi , que no has hecho el PR al fichero...

JJ reviewed (changes requested):
Retiro el OK hasta que no pase los tests en IV-21-22
